### PR TITLE
Sup-1173: metafield types

### PR DESCRIPTION
**Pull request by lujcheng**
What's changed
We are adding type support and accepting String, Boolean, Numbers (Int, Double, Float)

Why
Stay consistent React Native and Android SDK in their app.

---
### Checklist

- [x] The steps for distribution have been follow [here](https://www.notion.so/adasupport/Distribution-3a9dfc90c9b3449781b6f9c825f1dee8).
- [ ] A [Release Note](https://www.notion.so/adasupport/Creating-Release-Notes-36906dfa8a2b4f10a31dc23b8d46681a) has been drafted and published after merging to master.
- [x] PR has been labelled with its [impact level](https://www.notion.so/adasupport/Release-Management-Change-Definitions-c5a239ae075d4cc49bb1066f3e11f39f#b0af5e2c7bc7481a82303fa70b12e4f6)

### CODE REVIEW

**Comment by abhi-ada** [comment]
We should add deprecation warnings on the old methods of setting metafields.

**Comment by abhi-ada** [comment]
Also did we consider the initial setup? Do we need to change those APIs as well? On Android we did.

File: ExampleApp/ViewController.swift [hunk 1]
```diff
@@ -30,13 +30,14 @@ class ViewController: UIViewController {
             let firstName = firstNameField.text,
             let lastName = lastNameField.text,
             let email = emailField.text else { return }
+        let builder = MetaFields.Builder()
+        builder.setField(key: "firstName", value: firstName)
+            .setField(key: "lastName", value: lastName)
+            .setField(key: "email", value: email)
+        
+        adaFramework.setMetaFields(builder: builder)
```
[the remark this answers is not in the document]
Comment (lujcheng) [reply ExampleApp/ViewController.swift:39]: good call, I updated ExampleAppCocoapod as well

File: EmbedFramework/AdaWebHost.swift [hunk 2]
```diff
@@ -177,6 +200,71 @@ public class AdaWebHost: NSObject {
...
+            "sensitiveMetaFields": sensitiveMetaFields.build().metaFields,
+            "resetChatHistory": resetChatHistory
+        ]
+        guard let json = try? JSONSerialization.data(withJSONObject: data, options: .fragmentsAllowed),
+              let jsonString = String(data: json, encoding: .utf8) else { return }
+        
+        let toRun = "adaEmbed.reset(\(jsonString));"
+        
+        self.evalJS(toRun)
+    }
```
[the remark this answers is not in the document]
Comment (lujcheng) [reply EmbedFramework/AdaWebHost.swift:234]: I was following the android sdk example, where the reset function replaced metafield types to the builder as well. There are several reset overrides to resolve ambiguity issues with swift

File: ExampleAppCocoapod/ExampleAppCocoapod/ViewController.swift [hunk 3]
```diff
@@ -10,7 +10,7 @@ import UIKit
 import AdaEmbedFramework
 
 class ViewController: UIViewController {
-    lazy var adaFramework = AdaWebHost(handle: "nic", appScheme: "adaexampleapp", webViewLoadingErrorCallback: LoadingErrorCallback, webViewTimeout: 30.0)
+    lazy var adaFramework = AdaWebHost(handle: "coffees", appScheme: "adaexampleapp", webViewLoadingErrorCallback: LoadingErrorCallback, webViewTimeout: 30.0)
```
Comment (abhi-ada) [inline ExampleAppCocoapod/ExampleAppCocoapod/ViewController.swift:13]: "nic" is the bot handle we should use as default

**Review by abhi-ada — APPROVED** [review]

**Review by michaellazz — APPROVED** [review]

**Pull request merged**